### PR TITLE
Fix nex 1860/allow pdf assets to be packed on delivery compilation

### PR DESCRIPTION
**Pull request by ktsirkou**
Related to NEX-1860

Related PR: [extract-pdf-assets-on-export](https://github.com/oat-sa/extension-tao-itemqti/pull/1540)

**What is the issue**:
Having an item with any interaction that contains from Media asset import a PDF, we cannot export test or/and compile delivery.

**How to test.**

1. Install both this PR and the related one
2. Create an item with PDF media asset
3. Create a test with this item
4.
a) Export test
**Expected  behavior**: On export qti.xml of the item the PDF is reference as asset and not with taoMedia URI. On assets you can find the PDF.
b) Create a delivery
**Expected behavior**: The delivery is publishing successfully, Running the assessment the PDF is correctly presented

### CODE REVIEW

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

File: models/classes/pack/ItemPack.php [hunk 1]
```diff
@@ -41,13 +41,23 @@
...
+        'js',
+        'css',
+        'font',
+        'img',
+        'audio',
+        'video',
+        'xinclude',
+        'apip',
+        'pdf'
+    ];
```
Comment (shpran) [inline models/classes/pack/ItemPack.php:60]: Maybe a constant can be used here?
These values are static and do not change anywhere.

File: models/classes/pack/ItemPack.php [hunk 2]
```diff
@@ -205,15 +226,15 @@ public function JsonSerialize()
...
     }
 
     /**
      * @param array $assetEncoders
      */
-    public function setAssetEncoders($assetEncoders)
+    public function setAssetEncoders(array $assetEncoders): void
     {
         foreach ($assetEncoders as $type => $encoder) {
             if ($encoder == '') {
```
Comment (shpran) [inline models/classes/pack/ItemPack.php:240]:
```suggestion
            if ($encoder === '') {
```

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**